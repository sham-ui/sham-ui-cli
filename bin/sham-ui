#!/usr/bin/env node

require( 'commander' )
    .version( require( '../package' ).version )
    .usage( '<command> [options]' )
    .command( 'new', 'generate a new project from a template' ).alias( 'n' )
    .command( 'upgrade', 'upgrade existed project from a template' ).alias( 'u' )
    .command( 'generate', 'generate a new component' ).alias( 'g' )
    .command( 'collect', 'collect filters & directives' ).alias( 'c' )
    .command( 'destroy', 'destroy component' ).alias( 'd' )
    .parse( process.argv );