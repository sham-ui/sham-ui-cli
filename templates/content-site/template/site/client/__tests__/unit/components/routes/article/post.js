import setupUnsafe from 'sham-ui-unsafe';
import { createDI } from 'sham-ui';
import hrefto from 'sham-ui-router/lib/href-to';
import formatLocaleDate from '../../../../../src/filters/format-locale-date';
import RoutesArticlePost  from '../../../../../src/components/routes/article/post.sfc';
import renderer from 'sham-ui-test-helpers';

it( 'renders correctly', () => {
    const DI = createDI();
    setupUnsafe( DI );

    DI.bind( 'router', {
        generate: () => '/'
    } );

    const meta = renderer( RoutesArticlePost, {}, {
        DI,
        directives: {
            hrefto
        },
        filters: {
            formatLocaleDate
        }
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
