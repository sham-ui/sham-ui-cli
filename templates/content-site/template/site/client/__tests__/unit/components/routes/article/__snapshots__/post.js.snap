// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "Options": Object {
    "category": "",
    "content": "",
    "createdAt": "",
    "tags": Array [],
    "title": "",
  },
  "html": "
<div class=\\"container\\">
  <div class=\\"row\\">
    <div class=\\"col-lg-10 offset-lg-1 mb-20\\">
      <div class=\\"post-single\\">
        <div class=\\"post-single-content\\">
          <a class=\\"categorie\\" href=\\"/\\">null</a>
          <h4></h4>
          <div class=\\"post-single-info\\">
            <ul class=\\"list-inline\\">
              <li>Invalid Date</li>
            </ul>
          </div>
        </div>
        <div class=\\"post-single-body\\"></div>
        <div class=\\"post-single-footer\\">
          <div class=\\"tags\\">
            <ul class=\\"list-inline\\"></ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
",
}
`;
